libraries/core_libs/consensus/src/final_chain/final_chain.cpp

```diff
@@ -105,7 +105,7 @@ class FinalChainImpl final : public FinalChain {
           continue;
         }
         // Non-executed trxs
-        auto const& trx = new_blk.transactions[i];
+        auto& trx = new_blk.transactions[i];
```

Comment:
const auto trx? Since std::move(trx) later

Reply:
your suggestion is exactly the problem here - you cant move anything that is const, copy is used in such case...